platform/android/scripts/generate-style-code.js

```diff
@@ -59,20 +66,39 @@ global.propertyType = function propertyType(property) {
   }
 }
 
+global.propertyJavaType = function propertyType(property) {
+   switch (property.type) {
+       case 'boolean':
+         return 'boolean';
+       case 'number':
+         return 'float';
+       case 'string':
+         return 'String';
+       case 'enum':
+         return 'String';
+       case 'color':
+         return 'String';
+       case 'array':
+         return `${propertyJavaType({type:property.value})}[]`;
+       default:
+         throw new Error(`unknown type for ${property.name}`);
+   }
+ }
+
 global.propertyJNIType = function propertyJNIType(property) {
   switch (property.type) {
       case 'boolean':
         return 'jboolean';
-      case 'jfloat':
-        return 'Float';
+      case 'number':
+        return 'jfloat';
```

Comment:
I'm actually using that here: https://github.com/mapbox/mapbox-gl-native/pull/9013/files#diff-6cedd6a690e68c865421e25a5ccbb45dR47, might not be calling into the correct function?